starter_code/app.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const express = require('express');
+const app = express();
+const hbs = require('hbs');
+const path = require('path');
+const SpotifyWebApi = require('spotify-web-api-node');
+const credentials = require('./credentials.js');
+const spotifyApi = new SpotifyWebApi({
+  clientId: credentials.clientId,
+  clientSecret: credentials.clientSecret
+});
+const morgan = require('morgan');
+const fs = require('fs');
+
+// create a write stream (in append mode)
+const accessLogStream = fs.createWriteStream(path.join(__dirname, 'access.log'), { flags: 'a' });
+
+spotifyApi.clientCredentialsGrant()
+  .then(function (data) {
+    spotifyApi.setAccessToken(data.body['access_token']);
+    console.log('Spotify client credentials granted successfully');
+  }, function (err) {
+    console.log('Something went wrong when retrieving an access token', err);
+  });
+
+// Middleware
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'hbs');
+app.use(express.static(path.join(__dirname, 'public')));
+app.use(morgan('combined', { stream: accessLogStream }));
+
+// Routes
+app.get('/', (req, res, next) => {
+  res.render('index');
+});
+
+app.get('/artists', (req, res, next) => {
+  console.log('Search term: ', req.query.searchTerm);
+  spotifyApi.searchArtists(req.query.searchTerm)
+    .then(data => {
+      console.log(data.body.artists.items[0]);
+
+      res.render('artists', data);
+    })
+    .catch(err => {
+      // ----> 'HERE WE CAPTURE THE ERROR'
+      console.log(err);
+    });
+});
+
+app.get('/albums/:artistId', (req, res) => {
+  console.log('artist id:', req.params.artistId);
+
+  spotifyApi.getArtistAlbums(req.params.artistId).then(
+    function (data) {
+      console.log('Artist albums', data.body);
+      res.render('albums', data.body);
+    },
+    function (err) {
+      console.error(err);
+    }
+  );
+});
+
+app.get('/tracks/:albumId', (req, res) => {
+  spotifyApi.getAlbumTracks(req.params.albumId)
+    .then(function (data) {
+      console.log(data.body);
+      res.render('tracks', data.body);
+    }, function (err) {
+      console.log('Something went wrong!', err);
+    });
+});
+
+// Start listening
+app.listen(3000);
```

Comment:
Instead of app.listen, it's better to launch with the script on the package.json.